src/vos/vos_query.c

```diff
@@ -203,25 +198,53 @@ query_recx(struct open_query *query, daos_recx_t *recx)
 	if (rc != 0)
 		goto fini;
 
-	recx->rx_idx = entry.en_sel_ext.ex_lo;
-	recx->rx_nr = entry.en_sel_ext.ex_hi - entry.en_sel_ext.ex_lo + 1;
+	if (opc & EVT_ITER_SKIP_HOLES) {
+		if (entry.en_visibility & EVT_VISIBLE) {
```

Comment:
evt_iter_probe would have returned an error if there is no entry.  If it's not returning an error, it's a bug.  So you should not need this check.  You could assert it's visible if you want.